server/middleware/prepaids.coffee

```diff
@@ -116,3 +124,90 @@ module.exports =
+    token = req.body.stripe?.token
+    timestamp = req.body.stripe?.timestamp
+
+    if isNaN(maxRedeemers) or maxRedeemers < 1
+      throw new errors.UnprocessableEntity("Invalid number of licenses to buy: #{maxRedeemers}")
+
+    alreadyOwnedStarterLicenses = yield Prepaid.find({
+      creator: mongoose.Types.ObjectId(creator.id)
+      type: 'starter_license'
+    }).exec()
+    alreadyOwnedStarterLicenseCount = alreadyOwnedStarterLicenses.map((prepaid) -> prepaid.get('maxRedeemers')).reduce(((a,b) -> a+b), 0)
+
+    if maxRedeemers + alreadyOwnedStarterLicenseCount > Prepaid.MAX_STARTER_LICENSES
+      throw new errors.Forbidden('You cannot own more than 75 starter licenses.')
+    
+    if not (token or creator.isAdmin())
+      throw new errors.UnprocessableEntity('Missing required Stripe token')
+
+    if creator.isAdmin()
+      try
+        yield createStarterLicense({ creator: creator.id, maxRedeemers })
+        res.status(200).send(prepaid)
+      catch e
+        throw new errors.InternalServerError("Database error: #{e.message}")
+
+    else
+      product = yield Product.findOne({ name: 'starter_license' })
+
+      try
+        customer = yield StripeUtils.getCustomerAsync(creator, token)
+      catch e
+        logError(creator, "Stripe getCustomer error: #{JSON.stringify(err)}")
+      metadata =
+        type: 'starter_license'
+        userID: creator.id
+        timestamp: parseInt(timestamp)
+        maxRedeemers: maxRedeemers
+        productID: "prepaid starter_license"
+
+      totalAmount = maxRedeemers * product.get('amount')
+      try
+        charge = yield StripeUtils.createChargeAsync(creator, totalAmount, metadata)
+        prepaid = yield createStarterLicense({ creator: creator.id, maxRedeemers })
+        payment = yield StripeUtils.createPaymentAsync(creator, charge, {prepaidID: prepaid._id})
+        msg = "#{creator.get('email')} paid #{payment.get('amount')} for starter_license prepaid redeemers=#{maxRedeemers}"
+        slack.sendSlackMessage msg, ['tower']
+        res.status(200).send(prepaid)
+      catch err
+        logError(creator, "getCustomer error: #{JSON.stringify(err)}")
+        throw(err)
+
+createStarterLicense = wrap ({ creator, maxRedeemers }) ->
+  yield createPrepaid({
+    creator: creator
+    type: 'starter_license'
+    maxRedeemers, properties: {}
+    startDate: moment().toISOString()
+    endDate: moment().add(6, 'months').toISOString()
+    includedCourseIDs: STARTER_LICENSE_COURSE_IDS
+  })
+
+createPrepaid = wrap ({ creator, type, maxRedeemers, properties, startDate, endDate, includedCourseIDs }) ->
+  options =
+    creator: creator
+    type: type
+    code: yield Prepaid.generateNewCodeAsync()
+    maxRedeemers: parseInt(maxRedeemers)
+    properties: properties
+    redeemers: []
+    startDate: startDate
+    endDate: endDate
+    includedCourseIDs: includedCourseIDs
+  prepaid = new Prepaid(options)
+  yield prepaid.save()
+  return prepaid
```

Comment:
These two functions don't need to be wrapped. Prepaid.save() returns a thenable. So you can return that, which `createStarterLicense` will propagate up and `purchaseStarterLicenses` will yield. Basically when a function emits one promise at the end, you can just return that promise. Also, if you were to keep this as a generator function, you should use `co`'s wrap instead, since `wrap` is for `co-express`.